src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.UnrootedSymbolSet.cs

```diff
@@ -11,8 +11,27 @@
 
 namespace Microsoft.CodeAnalysis
 {
+    using SecondaryReferencedSymbol = (int hashCode, WeakReference<ISymbol> symbol, SolutionCompilationState.MetadataReferenceInfo referenceInfo);
+
     internal partial class SolutionCompilationState
     {
+        internal readonly record struct MetadataReferenceInfo(MetadataReferenceProperties Properties, string? FilePath)
+        {
+            internal static MetadataReferenceInfo From(MetadataReference reference)
+                => new(reference.Properties, (reference as PortableExecutableReference)?.FilePath);
+        }
+
+        /// <summary>
+        /// Information maintained for unrooted symbols. 
+        /// </summary>
+        /// <param name="ProjectId">
+        /// The project the symbol originated from, i.e. the symbol is defined in the project or its metadata reference.
+        /// </param>
+        /// <param name="ReferencedThrough">
+        /// If the symbol is defined in a metadata reference of <paramref name="ProjectId"/>, information about the reference.
```

Comment:
ah, thanks.

Reply:
so just to make sure, 'ReferencedThrough' must be a metadata-reference attached *directly* the Proejct instance (and also Compilation instance) for that ProjectId in this particular Solution snapshot.  Yes?  (Nite: consider doc'ing)

Reply:
Yes.